scoringrules/_interval.py

```diff
@@ -0,0 +1,140 @@
+import typing as tp
+from typing import Optional
+
+from scoringrules.backend import backends
+from scoringrules.core import interval
+
+if tp.TYPE_CHECKING:
+    from scoringrules.core.typing import Array, ArrayLike, Backend
+
+
+def interval_score(
+    observations: "ArrayLike",
+    lower: "Array",
+    upper: "Array",
+    alpha: "Array",
+    /,
+    axis: int = -1,
+    *,
+    backend: "Backend" = None,
+) -> "Array":
+    r"""Compute the Interval Score or Winkler Score [(Bracher J, Ray EL, Gneiting T, Reich NG, 2022)](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008618) for 1 - $\alpha$ prediction intervals PI = [lower, upper].
```

Comment:
Not sure if this is the most appropriate reference. @sallen12 ?

Reply:
I would probably use [this](https://doi.org/10.1198/016214506000001437) for the interval score and the current reference (with year 2021 instead of 2022) for the weighted interval score.

Reply:
Done :+1: